### ISSUE TITLE
Consul Configuration Server health check failling when using Vault->Consul backend

### ISSUE THREAD

**Issue by robscully** [spring-cloud/spring-cloud-config#565]
I have Spring Cloud Config server configured to use Vault as its backend. Vault is encrypting data correctly directly into Consul and I can configure my services directly from the Spring Cloud Config Server. Unfortunately, the health check for the Config Server in Consul is failing. The reason for this appears to be that Consul is not presenting the Config Server with the X-Config-Token which the Spring Config Server uses to query Vault. I can reproduce this issue with curl as follows
`curl -X "GET" http://localhost:8080/health
{"status":"DOWN","configServer":{"status":"DOWN","repository":{"application":"app","profiles":"default"},"error":"java.lang.IllegalArgumentException: Missing required header: X-Config-Token"},"discoveryComposite":{"description":"Spring Cloud Consul Discovery Client","status":"UP","discoveryClient":{"description":"Spring Cloud Consul Discovery Client","status":"UP","services":["consul"]}},"diskSpace":{"status":"UP","total":27345022976,"free":5925949440,"threshold":10485760},"refreshScope":{"status":"UP"},"consul":{"status":"UP","services":{"consul":[]},"advertiseAddress":"172.18.0.3","datacenter":"dc1","domain":"consul.","nodeName":"consulserver","bindAddress":"0.0.0.0","clientAddress":"0.0.0.0"},"hystrix":{"status":"UP"}}
`
 
Now should I try and curl while presenting the X-Config-Token the health check succeeds.

`curl -X "GET" http://localhost:8080/health -H "X-Config-Token: c2af647b-eba2-253c-7b6a-998c43b841ab"
{"status":"UP","configServer":{"status":"UP","repositories":[{"name":"app","profiles":["default"],"label":null}]},"discoveryComposite":{"description":"Spring Cloud Consul Discovery Client","status":"UP","discoveryClient":{"description":"Spring Cloud Consul Discovery Client","status":"UP","services":["consul"]}},"diskSpace":{"status":"UP","total":27345022976,"free":5924212736,"threshold":10485760},"refreshScope":{"status":"UP"},"consul":{"status":"UP","services":{"consul":[]},"advertiseAddress":"172.18.0.3","datacenter":"dc1","domain":"consul.","nodeName":"consulserver","bindAddress":"0.0.0.0","clientAddress":"0.0.0.0"},"hystrix":{"status":"UP"}}`

Is there a way to auto-configure Consul during registration to use this token to successfully complete the health check? I don't see any documentation mentioning this.

Or if there was a workaround for the config server to remove the repository health check it would probably suffice too since Conul performs its own health checks on Vault.

**Comment by spencergibb** [comment]
You can use `spring.cloud.config.server.health.enabled=false` to disable the health check.

**Comment by robscully** [comment]
Thanks Spencer.

Is there anyway to disable the repository check and keep the config server health check? I'd like to keep checking the config server health.

**Comment by spencergibb** [comment]
That's all the health check does.

**Comment by robscully** [comment]
Let me rephrase, as per my first Post the Healtg check returns a status of down for the repository. Is there any wat i can keep the health check and remove the repository from the results?

**Comment by spencergibb** [comment]
I understood. That is all the health check does. It would be a noop health check.

**Comment by robscully** [comment]
Okay. Is there anyway to get the Health check to pass so by having it pass the correct token?

**Comment by robscully** [comment]
The main issue this represents is for discover. If I want to use consul to discover my config server or consul dns then Consul will only return healthy services and for config server there would be none.

**Comment by spencergibb** [comment]
Not currently. Why would disabling the config server health check, make the config server not healthy to consul? Wouldn't `/health` still return OK?

**Comment by robscully** [comment]
Ah, okay j rhinknjnwas misunderstanding how the health check disabling worked! I'll test this out tomorrow and close out of it works - Thanks.

**Comment by robscully** [comment]
That was supposed to say:

Ah, okay I think I was misunderstanding how the health check disabling worked! I'll test this out tomorrow and close out of it works - Thanks.

**Comment by robscully** [comment]
That worked so closing this out.
